detox/src/devices/drivers/android/emulator/EmulatorDriver.js

```diff
@@ -51,7 +46,7 @@ class EmulatorDriver extends AndroidDriver {
     await this._avdValidator.validate(avdName);
     await this._fixEmulatorConfigIniSkinNameIfNeeded(avdName);
 
-    const adbName = await this._allocateDevice(avdName);
+    const adbName = await this.allocateDevice(avdName) || this._createDevice();
```

Comment:
@DmytroKost this might be the CI stability issue: `this._createDevice()` must be called from within `allocateDevice`, or the device wouldn't be locked... I'll try to fix that myself, in a bit...

Reply:
```suggestion
    const adbName = await this.allocateDevice(avdName);
```